core-web/apps/dotcms-ui/src/app/portlets/shared/dot-content-types-edit/components/dot-binary-settings/dot-binary-settings.component.ts

```diff
@@ -0,0 +1,200 @@
+    @Input() field: DotCMSContentTypeField;
+    @Input() isVisible: boolean = false;
+
+    @Output() changeControls = new EventEmitter<DotDialogActions>();
+    @Output() valid = new EventEmitter<boolean>();
+    @Output() save = new EventEmitter<DotFieldVariable[]>();
+
+    form: FormGroup;
+
+    private fb: FormBuilder = inject(FormBuilder);
+    private fieldVariablesService = inject(DotFieldVariablesService);
+    private dotMessageService = inject(DotMessageService);
+    private dotHttpErrorManagerService = inject(DotHttpErrorManagerService);
+
+    settingsMap = {
+        accept: {
+            key: 'accept',
+            variable: null
+        },
+        systemOptions: {
+            key: 'systemOptions',
+            variable: null
+        }
+    };
+
+    get settings() {
+        return Object.values(this.settingsMap);
+    }
+
+    protected readonly systemOptions = [
+        {
+            key: 'allowURLImport',
+            message: 'binary-field.settings.system.options.allow.url.import'
+        },
+        {
+            key: 'allowCodeWrite',
+            message: 'binary-field.settings.system.options.allow.code.write'
+        },
+        {
+            key: 'allowFileNameEdit',
+            message: 'binary-field.settings.system.options.allow.file.name.edit'
+        }
+    ];
+
+    ngOnChanges(changes: SimpleChanges) {
+        const { isVisible } = changes;
+        if (isVisible?.currentValue) {
+            this.changeControls.emit(this.dialogActions());
+        }
+    }
+
+    ngOnInit(): void {
+        this.form = this.fb.group({
+            accept: '',
+            // This is an object called systemOptions, this is going to the backend as a field variable
+            allowURLImport: false,
+            allowCodeWrite: false,
+            allowFileNameEdit: false
+        });
+
+        this.form.valueChanges.subscribe(() => {
+            this.valid.emit(this.form.valid);
+        });
+
+        this.fieldVariablesService
+            .load(this.field)
+            .subscribe((fieldVariables: DotFieldVariable[]) => {
+                fieldVariables.forEach((variable) => {
+                    const { key, value } = variable;
+
+                    if (key === 'accept') {
+                        this.settingsMap.accept.variable = variable;
+                        this.form.get(key)?.setValue(value);
+                    } else if (key === 'systemOptions') {
+                        this.settingsMap.systemOptions.variable = variable;
+                        const systemOptions = JSON.parse(value);
+
+                        this.systemOptions.forEach(({ key }) => {
+                            this.form.get(key)?.setValue(systemOptions[key]);
+                        });
+                    }
+                });
+            });
+    }
+
+    saveSettings(): void {
+        const updateActions = this.settings.map(({ variable, key }) => {
```

Comment:
Do we need the `settings` and `settingsMap`? We can have nested `FormGroup` so we can use this approach and let Angular handle the user interactions. We can also iterate over this to get the values. [Docs](https://angular.io/api/forms/FormGroupName)

```ts
const form = new FormGroup({
  systemOptions: new FormGroup({
    allowURLImport: new FormControl(''),
    allowFileNameEdit: new FormControl(''),
    allowCodeWrite: new FormControl(''),
  }),
  acceptType: new FormControl('')
});
```

Reply:
@dotCMS/frontend Have you ever tried this approach? To know if there is any trade-off we should be aware of

Reply:
As Jal returns, I'm helping to review the suggestions

This approach is good, but in the end an object like { key: 'systemOptions', value: 'VALUE_STRINGED'} is sent to the backend
Then we would also need to modify the value of the form, so adding a nested form would increase the complexity, since we will still have to transform it to adapt to the final object of the service

<img width="587" alt="Screenshot 2024-02-01 at 14 57 08" src="https://github.com/dotCMS/core/assets/144152756/8e310a6a-986c-4ac6-b51d-755b946bd63e">
<img width="588" alt="Screenshot 2024-02-01 at 14 57 13" src="https://github.com/dotCMS/core/assets/144152756/c55932bf-321c-496f-b918-8e9288314a34">

Reply:
The `clazz` and `fieldId` are constants belonging to the field, and can be received as `@Inputs`.

We don't need to modify the form value; instead, we can iterate over it and call the `JSON.stringify` when needed.

This is a quick example to illustrate how I think we should handle the form values:

```ts
const updatedVariables = Object.keys(this.form.controls).map(key => {
  const value = this.form.get(key).value;
  const variables = {
       key,
       clazz: this.clazz, // Constant because it belong to the field
       fieldId: this.fieldId, // Constant because it belong to the field
       value: value instanceof Object ? JSON.stringify(value) : value // It can return an object when using nested form
  }

return value
        ? this.fieldVariablesService.save(this.field, fieldVariable)
        : this.fieldVariablesService.delete(this.field, fieldVariable)
})
```